modules/extensions-spring-support/src/test/java/org/mule/module/extension/internal/MessageSourceTestCase.java

```diff
@@ -52,6 +58,21 @@ public void onException() throws Exception
         new PollingProber(TIMEOUT_MILLIS, POLL_DELAY_MILLIS).check(new JUnitLambdaProbe(() -> heisenberg.getMoney().longValue() == -1));
     }
 
+
+    @Test
+    public void enrichExceptionOnStart() throws Exception
+    {
+        Throwable exception = catchThrowable(() -> startFlow("sourceFailedOnStart"));
```

Comment:
I don't think to be necessary to catch the throwable, the only assertion is over the Exception it self. Maybe you can use the ExpectedException JUnit rule to check the type of the exception, the cause and the message.